OpenRA.Mods.Common/Activities/FindAndDeliverResources.cs

```diff
@@ -194,6 +198,21 @@ public override bool Tick(Actor self)
 					if ((loc - searchFromLoc.Value).LengthSquared > searchRadiusSquared)
 						return int.MaxValue;
 
+					// Add a cost modifier to harvestable cells to prefer resources that are closer to the refinery.
+					// This reduces the tendancy for harvesters to move in straight lines
+					if (procPos.HasValue && harvInfo.ResourceRefineryDirectionCostModifier > 0 && harv.CanHarvestCell(self, loc))
+					{
+						var pos = self.World.Map.CenterOfCell(loc);
+
+						// Calculate harv-cell-refinery angle (cosine rule)
+						var aSq = (harvPos - procPos.Value).LengthSquared;
+						var bSq = (pos - procPos.Value).LengthSquared;
+						var cSq = (pos - harvPos).LengthSquared;
+						var cosA = (int)(1024 * (bSq + cSq - aSq) / (2 * Exts.ISqrt(bSq * cSq)));
```

Comment:
That square root doesn't belong there.

Reply:
It does...? `Exts.ISqrt(bSq * cSq)` is equivalent to `b * c`, but without the redundant work of having to calculate `b` and `c` explicitly.

Reply:
ah alright, apparently I can't read.